hibernate-core/src/main/java/org/hibernate/jpa/event/internal/CallbackBuilderLegacyImpl.java

```diff
@@ -239,9 +237,22 @@ public void release() {
 
 	@SuppressWarnings({"unchecked", "WeakerAccess"})
 	public Callback[] resolveEmbeddableCallbacks(Class entityClass, Property embeddableProperty, CallbackType callbackType, ReflectionManager reflectionManager) {
-
-		final String embeddableClassName = embeddableProperty.getType().getReturnedClass().getName();
-		final XClass embeddableXClass = reflectionManager.classForName( embeddableClassName );
+		XClass embeddableXClass = null;
+		//First lookup for Collection of embeddable and get the embeddable class
+		XClass entityXClass = reflectionManager.classForName(entityClass.getName());
+		for (XProperty prop : entityXClass.getDeclaredProperties(AccessType.FIELD.getType())) {
+			if (prop.getName().equals(embeddableProperty.getName())) {
+				if(prop.isCollection()){
```

Comment:
Seems we can combine the above two if together

Reply:
Also, I'd make the `prop.isCollection` check the primary - perf.  Its a check against a boolean field versus char-by-char comparison of a String.

Reply:
Good point. Thanks. Will put it in my toolbox for my future PRs.